<a href="https://colab.research.google.com/github/Vatsalp599/Vatsalp599-DataScience-GenAI-submissions/blob/main/EDA_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

I asked AI " " want you to code and then when i run it i should get option of uploading the file and create dataframe"

In [ ]:
import pandas as pd
from google.colab import files
import io

print("Please upload your file:")
uploaded = files.upload()

for fn in uploaded.keys():
  print('User uploaded file "{name}" with length {length} bytes'.format(
      name=fn, length=len(uploaded[fn])))

  # Assuming the uploaded file is a CSV
  df = pd.read_csv(io.StringIO(uploaded[fn].decode('utf-8')))
  print(f"DataFrame '{fn}' created successfully:")
  display(df.head())

Please upload your file:


Saving EDA-DP.csv to EDA-DP.csv
User uploaded file "EDA-DP.csv" with length 948117 bytes
DataFrame 'EDA-DP.csv' created successfully:


work_year experience_level employment_type                  job_title  \
0       2024               SE              FT                AI Engineer   
1       2024               SE              FT                AI Engineer   
2       2024               SE              FT              Data Engineer   
3       2024               SE              FT              Data Engineer   
4       2024               SE              FT  Machine Learning Engineer   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0  202730             USD         202730                 US             0   
1   92118             USD          92118                 US             0   
2  130500             USD         130500                 US             0   
3   96000             USD          96000                 US             0   
4  190000             USD         190000                 US             0   

  company_location company_size  
0               US            M  
1               US            M  
2               US            M  
3               US            M  
4               US            M

As expected, it gave me an option to upload file, which i did and whose Data head it gave out, which seems alright as we can see the column names and first five entries

Now i asked AI to " please give me the information about the data using the info function"

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16534 entries, 0 to 16533
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           16534 non-null  int64 
 1   experience_level    16534 non-null  object
 2   employment_type     16534 non-null  object
 3   job_title           16534 non-null  object
 4   salary              16534 non-null  int64 
 5   salary_currency     16534 non-null  object
 6   salary_in_usd       16534 non-null  int64 
 7   employee_residence  16534 non-null  object
 8   remote_ratio        16534 non-null  int64 
 9   company_location    16534 non-null  object
 10  company_size        16534 non-null  object
dtypes: int64(4), object(7)
memory usage: 1.4+ MB


As in the output we can see there are total of 16534 rows and all the columns have same with no missing values and we can also see the data types. Columns with object dtype usually contain string-based or mixed data, which often represent categorical variables.Columns with int64 dtype contain integer numerical values.

to double check specifically for mising data i asekd AI " find missing values in all columns"

In [ ]:
df.isnull().sum()

work_year             0
experience_level      0
employment_type       0
job_title             0
salary                0
salary_currency       0
salary_in_usd         0
employee_residence    0
remote_ratio          0
company_location      0
company_size          0
dtype: int64

As expected, the output shows no missing values

Now to check for number of duplicate and unique rows i asked AI " count the number of duplicate rows in our dataset"

In [ ]:
df.duplicated().sum()

np.int64(6421)

As i can see that the output gives me that there are 6421 duplicate rows in our dataset, but we cannot reomve them unless we know they are exact duplicates of other row and not provide us any additional information

now i asked AI " please state the exact duplicate rows with the orignal row to check"

In [ ]:
display(exact_duplicates)

work_year experience_level employment_type           job_title  salary  \
16296       2021               MI              FT       Data Engineer  200000   
16376       2021               MI              FT       Data Engineer  200000   
16452       2021               MI              FT      Data Scientist   76760   
16490       2021               MI              FT      Data Scientist   76760   
14839       2022               EN              FT        Data Analyst   50000   
...          ...              ...             ...                 ...     ...   
5638        2024               SE              FT  Research Scientist  440000   
5662        2024               SE              FT  Research Scientist  440000   
5796        2024               SE              FT  Research Scientist  440000   
2635        2024               SE              FT  Research Scientist  500000   
4216        2024               SE              FT  Research Scientist  500000   

      salary_currency  salary_in_usd employee_residence  remote_ratio  \
16296             USD         200000                 US           100   
16376             USD         200000                 US           100   
16452             EUR          90734                 DE            50   
16490             EUR          90734                 DE            50   
14839             USD          50000                 US            50   
...               ...            ...                ...           ...   
5638              USD         440000                 US             0   
5662              USD         440000                 US             0   
5796              USD         440000                 US             0   
2635              USD         500000                 US           100   
4216              USD         500000                 US           100   

      company_location company_size  
16296               US            L  
16376               US            L  
16452               DE            L  
16490               DE            L  
14839               US            L  
...                ...          ...  
5638                US            M  
5662                US            M  
5796                US            M  
2635                US            M  
4216                US            M  

[8973 rows x 11 columns]

As we can see some rows are exact duplicates of each other, hence it does not provide any additional information so we should delete them

to delete exact duplicates i asked AI " please delete all teh rows with exact duplicates and only keep orignal ones in the dataset"

In [ ]:
df = df.drop_duplicates(keep='first')

now as the code it run only to keep first and delete all exact duplicates, now lets check duplicates again by asking AI " check for duplicate rows in our dataset"

In [ ]:
df.duplicated().sum()

np.int64(0)

As we can see our code that AI gave out worked and now our dataset does not have any duplicates